drivers/tee/optee/optee.c

```diff
@@ -226,12 +392,21 @@ static int optee_call(const struct device *dev, struct optee_msg_arg *arg)
 		data->smc_call(param.a0, param.a1, param.a2, param.a3,
 			       param.a4, param.a5, param.a6, param.a7, &res);
 
+		/*
+		 * Clean allocated pages if needed. Some function calls requires pages
+		 * allocation which should be freed after processing new request
+		 */
+		if (pages) {
```

Comment:
How you know that OP-TEE finished reading from list already?

Reply:
good point. I'll take a look on it. Thanks

Reply:
It is safe to free this list only after you get another SHM op (e,g. another alloc or free)